internal/junit/junit.go

```diff
@@ -88,6 +109,58 @@ type TestSuite struct {
 	SystemOut string     `xml:"system-out,omitempty"`
 }
 
+// AddTestCases adds test cases to the test suite. If unique is true, existing
+// test cases with the same name will be replaced.
+func (ts *TestSuite) AddTestCases(unique bool, tcs ...TestCase) {
+	if !unique {
+		ts.TestCases = append(ts.TestCases, tcs...)
+		return
+	}
+
+	// index existing test cases by name
+	testMap := make(map[string]TestCase)
+	for _, tc := range ts.TestCases {
+		key := fmt.Sprintf("%s.%s", tc.ClassName, tc.Name)
+		testMap[key] = tc
+	}
+
+	// add new test cases
+	for _, tc := range tcs {
+		key := fmt.Sprintf("%s.%s", tc.ClassName, tc.Name)
+		testMap[key] = tc
+	}
+
+	// convert map back to slice
+	ts.TestCases = maps.Values(testMap)
+}
+
+// Compute updates some statistics for the test suite based on the test cases it
+// contains.
+//
+//		Updates the following fields:
```

Comment:
The format looks weird. Two tabs?

Reply:
Whoops! My IDE was auto formatting the extra spacing I was adding (to ensure that the list shows up as as an actual list in the IDE docs) to tabs on save 🤦